tests/test_types.ts

```diff
@@ -0,0 +1,28 @@
+import RE2 from 're2';
+
+function assertType<T>(_val: T) {}
+
+function test_execTypes() {
+  const re = new RE2('quick\\s(brown).+?(?<verb>jumps)', 'ig');
+  const result = re.exec('The Quick Brown Fox Jumps Over The Lazy Dog')
+  if (!(result && result.groups)) {
+    throw 'Unexpected Result'
+  }
+  assertType<number>(result.index)
+  assertType<string>(result.input)
+  assertType<string | undefined>(result.groups['verb'])
+}
+
+function test_matchTypes() {
+  const re = new RE2('quick\\s(brown).+?(?<verb>jumps)', 'ig');
+  const result = re.match('The Quick Brown Fox Jumps Over The Lazy Dog')
+  if (!(result && result.index && result.input && result.groups)) {
+    throw 'Unexpected Result'
+  }
+  assertType<number>(result.index)
+  assertType<string>(result.input)
+  assertType<string | undefined>(result.groups['verb'])
+}
+
+test_execTypes()
+test_matchTypes()
```

Comment:
I don't think this file is ever executed. I added `console.log()` and I couldn't see its output. I tried to run it with `ts-node` and it fails to compile (some import problems).

Reply:
Yes. These code won't run. However TypeScript checks them with tsc. Removing them makes following error with current config.

```console
❯ npx tsc
tests/test_types.ts:16:10 - error TS6133: 'test_matchTypes' is declared but its value is never read.

16 function test_matchTypes() {
            ~~~~~~~~~~~~~~~


Found 1 error in tests/test_types.ts:16
```